### PR TITLE
Bugfix: Issue #97 - Ignore add/change messages from Chokidar on files that don't match the pattern.

### PR DESCRIPTION
**Pull request by samuelneff**
tl;dr

* Works around a bug in a dependency
* No change in behavior other than fixing the bug
* No change in API or documentation or examples

[Chokidar has a bug](https://github.com/paulmillr/chokidar/issues/967) where it does not filter paths by the provided pattern if they are inside a symlink. The bug is two years old. 

This PR rechecks the file path against the pattern if we're watching to work around the Chokidar bug.

It adds a dependency on [minimatch](https://www.npmjs.com/package/minimatch) but that dependency already existed via a transient dependency (actually several versions are referenced currently due to already existing incorrect transient dependencies on glob in some reporters). 

Chokidar issue:

  https://github.com/paulmillr/chokidar/issues/967

When that's fixed this PR can be reverted, but the issue is 2 years old already (reported 2020).

### CODE REVIEW

**Comment by samuelneff** [comment]
**Update**: I am submitting a PR to Chokidar for the underlying issue. If that gets merged then this PR can be replaced with a dependency update instead. That'd be better.
